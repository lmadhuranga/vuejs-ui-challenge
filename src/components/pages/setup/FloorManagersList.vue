<style scoped>
  .panelMedium {
    position: fixed;
    top: 0;
    background: #fff;
    max-width: 600px;
    height: 100vh;
    right: 0;
    width: 100%;
  }

  .panelMedium.active:after {
    display: block;
    content: " ";
    position: fixed;
    width: 100%;
    height: 100%;
    background: rgba(15, 4, 35, 0.63);
    right: 600px;
    top: 0;
    z-index: -15;
  }

  .panelMedium .accordionBody {
    /*padding: 20px;*/
    display: none;
  }

  .panelMedium .accordionBody.active {
    display: block;
  }

  .panelMedium .accordionHeader {
    padding: 20px;
  }

  .panelMedium .panelClose {
    position: absolute;
    margin-left: -46px;
    margin-top: 15px;
    padding: 7px 10px;
    background: #fff;
    border-radius: 100%;
  }

  .panelMedium .tabHeaderCollection a {
    color: #9E9E9E;
    text-decoration: none;
    padding-bottom: 10px;
    border-bottom: 2px solid;
    width: 100%;
    text-align: center;
  }

  .panelMedium .tabHeaderCollection a.active {
    color: #7c2bad;
  }

  .panelMedium .tabHeaderCollection {
    display: flex;
    justify-content: space-evenly;
  }

  .panelMedium .helpText {
    margin-top: 20px;
    display: inline-block;
  }
</style>
<template>
  <div id="floorManagersList" class="panelMedium active">
    <a class="panelClose">X</a>
    <div class="accordion">
      <navigator msg="Go To Home" path="/setup/"></navigator>
      <div class="accordionBody active">
        <tab-header-collection :currentRoute="currentRoute"></tab-header-collection>
        <div class="tabBodyCollection">

          <!-- s tab Zoeken -->
          <div class="tabBody active">
            <div class="wall">
              <h2>Floor Managers</h2>
              <floor-managers-list-menu :floorManagers="users"></floor-managers-list-menu>
            </div>
          </div>
          <!-- e tab Zoeken -->
        </div>
      </div>
    </div>
  </div>
</template>
<script>
  import FloorManagersListMenu from "../../template/FloorManagersListMenu";
  import TabHeaderCollection from "../../template/TabHeaderCollection";
  import Navigator from "../../template/Navigator";

  export default {
    components: {
      Navigator,
      TabHeaderCollection,
      FloorManagersListMenu,
    },
    name:'floorManagersList',
    props: ['users'],
    data(){
        return {
          currentRoute:'floorManagersList',
        }
    }

  }
</script>
