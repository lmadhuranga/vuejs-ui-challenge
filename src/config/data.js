export default {
  users: [
    {
      id: 1,
      name: 'Madhuranga Sendheera',
      address: 'Colombo, Sri Lanka',
      postCode: '12344',
      btwNumber: '12345',
      kvkNumber: '12345'
    },
    {
      id: 2,
      name: 'Randika Senadheera',
      address: 'Colombo, Sri Lanka',
      postCode: '12344',
      btwNumber: '12345',
      kvkNumber: '12345'
    },
  ],
  projects: {
    1: {
      id: 1,
      name: 'Project 1',
      shifts: [
        {
          id: 1,
          name: 'shift1',
          iban: 'IbanDetails',
          description: 'description',
          userId: 1
        },
        {
          id: 2,
          name: 'shift2',
          iban: 'IbanDetails',
          description: 'description',
          userId: 2
        }
      ]
    },
    2: {
      id: 1,
      name: 'Project 2'
    },
    3: {
      id: 1,
      name: 'Project 3'
    }
  }
}
