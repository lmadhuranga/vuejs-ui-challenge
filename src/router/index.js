import Vue from 'vue'
import Router from 'vue-router'
import Home from '@/components/pages/HomePage'
import UserAdd from '@/components/pages/UserAdd'
import UserView from '@/components/pages/UserView'
import UsersList from '@/components/pages/UsersList'
import SetupContainer from '@/components/pages/SetupContainer'
import Vuetify from 'vuetify'

Vue.use(Vuetify);
Vue.use(Router);

if (typeof window !== 'undefined' && window.Vue) {
  window.Vue.use(plugin)
}

export default new Router({
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home
    },
    {
      path: '/userslist',
      name: 'usersList',
      component: UsersList
    },
    {
      path: '/setup',
      name: 'setup',
      component: SetupContainer,
      children: [
        {
          // UserProfile will be rendered inside User's <router-view>
          // when /user/:id/profile is matched
          path: 'add',
          component: UserAdd
        }]
    },
    {
      path: '/add',
      name: 'add',
      component: UserAdd
    },
    {
      path: '/:id',
      name: 'userView',
      component: UserView
    }
  ]
})

