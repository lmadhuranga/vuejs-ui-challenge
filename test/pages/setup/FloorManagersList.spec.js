import Vue from 'Vue';
import FloorManagersList from '../../../src/components/pages/setup/FloorManagersList.vue'
import router from '../../../src/router'
import data from '../../../src/config/data'
import {shallow, mount} from "vue-test-utils"
import VueRouter from 'vue-router';

describe('FloorManagersList.vue', () => {
  Vue.use(VueRouter);
  let cmp, users;

  const createCmp = propsData => mount(FloorManagersList, {
    propsData,
    el: document.createElement('div'),
    router: router,
    render: h => h('router-view')
  });

  beforeEach(() => {
    users = data.users;
    cmp = createCmp({
      users: users
    });
  });

  it('should be able load floorManagers', () => {
    expect(cmp.vm.users).toEqual(users);
  });

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  });
});
