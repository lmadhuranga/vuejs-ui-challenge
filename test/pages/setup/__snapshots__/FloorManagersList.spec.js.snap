// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloorManagersList.vue has the expected html structure 1`] = `
<v-app
  toolbar--fixed=""
  toolbar=""
  style="overflow: hidden;"
>
  <main
    id="scrolling-techniques"
    class="scroll-y"
    style="max-height: 600px;"
  >
    <h2>
      Home Page
    </h2>
    <v-form
      lazy-validation=""
    >
      <v-text-field
        label="Name"
        rules="function (v) {
              cov_1gh2s8ykqq.f[1]++;
              cov_1gh2s8ykqq.f[1]++;
              cov_1gh2s8ykqq.s[1]++;
              cov_1gh2s8ykqq.s[1]++;
              return (cov_1gh2s8ykqq.b[0][0]++, (cov_1gh2s8ykqq.b[0][0]++, !!v)) || (cov_1gh2s8ykqq.b[0][1]++, (cov_1gh2s8ykqq.b[0][1]++, 'Name is required'));
            },function (v) {
              cov_1gh2s8ykqq.f[2]++;
              cov_1gh2s8ykqq.f[2]++;
              cov_1gh2s8ykqq.s[2]++;
              cov_1gh2s8ykqq.s[2]++;
              return (cov_1gh2s8ykqq.b[1][0]++, (cov_1gh2s8ykqq.b[1][0]++, v)) && (cov_1gh2s8ykqq.b[1][1]++, (cov_1gh2s8ykqq.b[1][1]++, v.length <= 10)) || (cov_1gh2s8ykqq.b[1][2]++, (cov_1gh2s8ykqq.b[1][2]++, 'Name must be less than 10 characters'));
            }"
        counter="10"
        required="required"
      />
      <v-text-field
        label="E-mail"
        rules="function (v) {
              cov_1gh2s8ykqq.f[3]++;
              cov_1gh2s8ykqq.f[3]++;
              cov_1gh2s8ykqq.s[3]++;
              cov_1gh2s8ykqq.s[3]++;
              return (cov_1gh2s8ykqq.b[2][0]++, (cov_1gh2s8ykqq.b[2][0]++, !!v)) || (cov_1gh2s8ykqq.b[2][1]++, (cov_1gh2s8ykqq.b[2][1]++, 'E-mail is required'));
            },function (v) {
              cov_1gh2s8ykqq.f[4]++;
              cov_1gh2s8ykqq.f[4]++;
              cov_1gh2s8ykqq.s[4]++;
              cov_1gh2s8ykqq.s[4]++;
              return (cov_1gh2s8ykqq.b[3][0]++, (cov_1gh2s8ykqq.b[3][0]++, /^\\w+([\\.-]?\\w+)*@\\w+([\\.-]?\\w+)*(\\.\\w{2,3})+$/.test(v))) || (cov_1gh2s8ykqq.b[3][1]++, (cov_1gh2s8ykqq.b[3][1]++, 'E-mail must be valid'));
            }"
        required="required"
      />
      <v-text-field
        label="About"
        multi-line=""
      />
      <v-btn
        class="btn-add-user"
      >
        Add User
      </v-btn>
      <a
        href="#/setup/projectsList"
        class=""
      >
        Select Floor Manager
      </a>
    </v-form>
  </main>
</v-app>
`;
